src/jit/morph.cpp

```diff
@@ -13183,7 +13191,7 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                         // if negative negate (min-int does not need negation)
                         if (mult < 0 && mult != SSIZE_T_MIN)
                         {
-                            tree->gtOp.gtOp1 = op1 = gtNewOperNode(GT_NEG, op1->gtType, op1);
+                            tree->gtOp.gtOp1 = op1 = gtNewOperNode(GT_NEG, tree->gtType, op1);
```

Comment:
I don't think this is technically necessary because this node always feeds into the `GT_MUL` created below, which always assumes its operand is already widened. However from the explanations in #18238 I assume this is still preferable.